sample-app/src/service-worker.ts

```diff
@@ -0,0 +1,71 @@
+/// <reference lib="webworker" />
+import { clientsClaim } from 'workbox-core';
+import { precacheAndRoute } from 'workbox-precaching';
+
+declare const self: ServiceWorkerGlobalScope;
+
+clientsClaim();
+const precacheManifest = self.__WB_MANIFEST;
+// Only precache static assets in production.
+// Precaching in development can result in confusing situations.
+// See https://github.com/facebook/create-react-app/issues/2398#issuecomment-304638935
+if (process.env.NODE_ENV !== 'development') {
+  precacheAndRoute(precacheManifest);
+}
+
+const cacheName = 'YEXT_ANSWERS_CACHE';
+const universalSearch = /v2\/accounts\/me\/answers\/query/;
+const universalAutocomplete = /v2\/accounts\/me\/answers\/autocomplete/;
+const verticalAutocomplete = /v2\/accounts\/me\/answers\/vertical\/autocomplete/;
+const swFetchedOnHeader = 'yxt-sw-fetched-on';
+
+// Clear the cache when a new service worker activates
+self.addEventListener('activate', event => {
+  event.waitUntil(caches.delete(cacheName));
+});
+
+self.addEventListener('fetch', (event: FetchEvent) => {
+  // Only cache universal search requests intended for VisualAutocomplete
+  if (
+    universalSearch.test(event.request.url) &&
+    new URLSearchParams(event.request.url).has('restrictVerticals')
+  ) {
+    event.respondWith(respondUsingCache(event.request, 30 * 1000));
+  }
+
+  if ([universalAutocomplete, verticalAutocomplete].find(regex => regex.test(event.request.url))) {
+    event.respondWith(respondUsingCache(event.request, 30 * 1000));
+  }
+});
+
+async function respondUsingCache(request: Request, timeToLiveInMilliseconds: number): Promise<Response> {
+  const cachedResponse = await caches.match(request);
+  const fetchedOn = cachedResponse?.headers.get(swFetchedOnHeader);
```

Comment:
Could we give this a more descriptive name? I'm not really sure what `fetchedOn` means

Reply:
renamed to `fetchedOnTimestampHeader` if that works?